tools/qontract_cli.py

```diff
@@ -1199,6 +1202,148 @@ def template(ctx, cluster, namespace, kind, name):
+            key, value = al.split("=")
+        except ValueError:
+            print(f"Additional label {al} not passed using key=value format")
+            sys.exit(1)
+
+        if not key or not value:
+            print(f"Additional label {al} not passed using key=value format")
+            sys.exit(1)
+
+        additional_labels[key] = value
+
+    gqlapi = gql.get_api()
+    namespaces = gqlapi.query(orb.NAMESPACES_QUERY)["namespaces"]
+    cluster_namespaces = [n for n in namespaces if n["cluster"]["name"] == cluster]
+
+    if len(cluster_namespaces) == 0:
+        print(f"Unknown cluster {cluster}")
+        sys.exit(1)
+
+    settings = queries.get_app_interface_settings()
+    if secret_reader == "config":
+        settings["vault"] = False
+    else:
+        settings["vault"] = True
+
+    # Get alertmanager config
+    am_config = ""
+    for ni in cluster_namespaces:
+        if ni["name"] != alertmanager_namespace:
+            continue
+        for r in ni.get("openshiftResources"):
+            if r["path"] != alertmanager_secret_path:
+                continue
+            openshift_resource = orb.fetch_openshift_resource(r, ni, settings)
+            body = openshift_resource.body
+            if "data" in body:
+                am_config = base64.b64decode(
+                    body["data"][alertmanager_secret_key]
+                ).decode("utf-8")
+            elif "stringData" in body:
+                am_config = body["stringData"][alertmanager_secret_key]
+            else:
+                print("Cannot get alertmanager configuration")
+                sys.exit(1)
+        break
+
+    rule_spec = {}
+    for ni in cluster_namespaces:
+        if ni["name"] != namespace:
+            continue
+        for r in ni.get("openshiftResources"):
+            if r["path"] != rules_path:
+                continue
+            openshift_resource = orb.fetch_openshift_resource(r, ni, settings)
+            if openshift_resource.kind.lower() != "prometheusrule":
+                print(f"Object in {rules_path} is not a PrometheusRule")
+                sys.exit(1)
+            rule_spec = openshift_resource.body["spec"]
+
+            break  # openshift resource
+        break  # cluster_namespaces
+
+    if not rule_spec:
+        print(
+            f"Cannot find any rule in {rules_path} from cluster {cluster} and "
+            f"namespace {namespace}"
+        )
+        sys.exit(1)
+
+    labels = []  # array of dicts
+    for group in rule_spec["groups"]:
+        for rule in group["rules"]:
+            if rule["alert"] != alert_name:
+                continue
+            labels.append(rule["labels"] | additional_labels)
+
+    if not labels:
+        print(f"Cannot find alert {alert_name} in rules {rules_path}")
+        sys.exit(1)
+
+    for lbl in labels:
+        result = amtool.config_routes_test(am_config, lbl)
+        if not result:
+            print(f"Error running amtool: {result}")
+        print(result)
```

Comment:
should this only be printed on success? Currently, if there is an error, we print result 2 times in a row.

Reply:
Fixed in 5def72c9